### PR TITLE
feat: Lt-2147 - ultimate recipients on open licence

### PR DESCRIPTION
**Pull request by Mark-Spiers-kainos**
This story allows for ultimate recipient to be added to open military/dual-use applications

### CODE REVIEW

File: applications/creators.py [hunk 1]
```diff
@@ -126,7 +126,7 @@ def _validate_goods_types(draft, errors, is_mandatory):
     return errors
 
 
-def _validate_ultimate_end_users(draft, errors, is_mandatory):
+def _validate_ultimate_end_users(draft, errors, is_mandatory, goods_type=False):
```
Comment (andytaylorkainos) [inline applications/creators.py:129]: id prefer if the boolean was replaced with the application subtype (standard or open)

File: conf/decorators.py [hunk 2]
```diff
@@ -161,3 +163,34 @@ def inner(request, *args, **kwargs):
...
+
+            if application.case_type.sub_type == CaseTypeSubTypeEnum.OPEN and party_type:
+                if (
+                    application.goodstype_category == GoodsTypeCategory.MILITARY
+                    and party_type == PartyType.ULTIMATE_END_USER
+                ):
+                    pass
+                else:
+                    return JsonResponse(
+                        data={"errors": ["This type of party can not be added to this type of open application"]},
```
Comment (andytaylorkainos) [inline conf/decorators.py:188]: content

File: applications/tests/tests_goods_type_on_application.py [hunk 3]
```diff
@@ -56,7 +56,7 @@ def test_create_goodstype_multiple_clcs_on_open_application_as_exporter_user_suc
         self.assertEquals(response_data["description"], "Widget")
         self.assertEquals(response_data["is_good_controlled"], True)
         self.assertEquals(
-            response_data["control_list_entries"],
+            sorted(response_data["control_list_entries"], key=lambda i: i["rating"]),
```
Comment (Mark-Spiers-kainos) [inline applications/tests/tests_goods_type_on_application.py:59]: this fixes one of the 2 api tests that randomly break.

File: applications/creators.py (hunk 1 above)
Comment (Mark-Spiers-kainos) [reply applications/creators.py]: changed the boolean name with "open_application" defaults to False keeping all logic the same below.

File: applications/creators.py [hunk 4]
```diff
@@ -344,6 +350,8 @@ def _validate_open_licence(draft, errors):
     errors = _validate_end_use_details(draft, errors, draft.case_type.sub_type)
     errors = _validate_temporary_export_details(draft, errors)
     errors = _validate_route_of_goods(draft, errors)
+    if draft.goodstype_category == "military":
```
Comment (stephanieboyd) [inline applications/creators.py:353]: Is this a constant somewhere?

**Pull request merged**